trillian/ctfe/services.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctfe
+
+import (
+	"context"
+	"crypto/sha256"
+
+	"github.com/google/certificate-transparency-go/trillian/ctfe/cache"
+	"github.com/google/certificate-transparency-go/trillian/ctfe/storage"
+	"k8s.io/klog/v2"
+)
+
+type IssuanceChainStorageBackend string
+
+const (
+	IssuanceChainStorageBackendTrillianGRPC = IssuanceChainStorageBackend("TrillianGRPC")
+	IssuanceChainStorageBackendCTFE         = IssuanceChainStorageBackend("CTFE")
+)
+
+type IssuanceChainService struct {
+	storageBackend IssuanceChainStorageBackend
```

Comment:
Now that the query methods have been removed, this field can be removed from the struct and from `NewIssuanceChainService` below. We could also remove the definitions of the consts from this PR.

Reply:
Those consts are the mapping of the generated protobuf enum. I can remove them in this PR but it is part of the constructor. I feel hesitate to change the constructor in the next PR (maybe I should't).

Shall we use the generated protobuf enum consts or our defined and mapped consts in the source code (e.g. instance.go)?

Reply:
Having multiple definitions of the same enums generally leads to a bad time. Best to have a single canonical definition. As this already exists in protbuf, then I recommend to use that. This is what we do in Trillian:

https://github.com/google/trillian/blob/master/storage/mysql/log_storage.go#L334

Reply:
That would be a long one with enum (i.e. `configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_TRILLIAN_GRPC`) but I think it should be fine.